### PR TITLE
build packages for python 3.10

### PR DESCRIPTION
**Pull request by proppy**
- github/workflow: add py310 entries
- openroad: use PYTHON_VERSION
- klayout: use pin_compatible
- openlane: update test design
- misc/open_pdks: move python to build dep
- misc/netgen: remove python dep

Related #343 #72

### CODE REVIEW

**Comment by proppy** [comment]
@ajelinski @kgugala @umarcor does that sounds like the right thing to do?

File: pnr/openroad/meta.yaml [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # Use `conda-build-prepare` before building for a better version string.
 {% set version = '%s_%04i_%s'|format(GIT_DESCRIBE_TAG|replace('v', '')|replace('-', '')|default('0.X'), GIT_DESCRIBE_NUMBER|int, GIT_DESCRIBE_HASH|default('gUNKNOWN')) %}
+
+{% set python_version = PYTHON_VERSION | default('3.7') %}
```
Comment (ajelinski) [inline pnr/openroad/meta.yaml:4]: Please add `py_suffix` too:
```
{% set py_suffix = 'py%s'|format(python_version|replace('.', '')) %}
```
and use it for build string in lines 25-26:
```
  # string: 20220203_1935_py37
  string: {{ environ.get('DATE_STR') }}_{{ py_suffix }}
```

File: .github/workflows/Build.yml [hunk 2]
```diff
@@ -313,7 +324,19 @@ jobs:
...
+      - uses: ./ci
+
+  symbiyosys-linux-py310:
+    needs: ["yosys-linux-py310"]
+    runs-on: [self-hosted, Linux, X64]
+    container: ubuntu:22.04
+    env:
+      PACKAGE: "formal/symbiyosys"
+      OS_NAME: "linux"
+      PYTHON_VERSION: "3.10"
```
Comment (ajelinski) [inline .github/workflows/Build.yml:339]: I see SymbiYosys still can't be built successfully so I'd say let's keep `SKIP` for jobs with Python 3.7/3.8. Unless building is fixed I'm not sure if adding this job for SymbiYosys+Python 3.10 makes sense. It'd be best to fix it but if not maybe let's add the job with `SKIP: "true"`?

Comment (proppy) [reply .github/workflows/Build.yml:339]: Understood, IIRC build failure used to block publication but it doesn't seem to be the case anymore. Maybe we could revisit removing skipped build in a follow up changes (and let them fails until they are fixed)

Comment (ajelinski) [reply .github/workflows/Build.yml:339]: Right, it makes sense to just let them fail. Some of them might have been even fixed upstream already so maybe you could drop all `SKIP`s in a separate commit then?

**Comment by proppy** [comment]
@ajelinski does this look good to you?

**Comment by ajelinski** [comment]
> @ajelinski does this look good to you?

The changes strictly related to the Python 3.10 addition LGTM.

Can't say much about other changes but if the packages are built correctly with such requirement changes, `open_pdks.sky130a` needs the new patch and `openlane` should be tested with `-design spm` instead of `-design inverter` then also LGTM.

**Comment by proppy** [comment]
@ajelinski do you prefer that I split those in separate PRs?

The issue is that I couldn't really test those, since they were all related (we can't test python 3.10 without design changes for broken packages, and we can't test the design changes without python 3.10 fixes because of dep conflicts)

**Comment by ajelinski** [comment]
> @ajelinski do you prefer that I split those in separate PRs?
> 
> The issue is that I couldn't really test those, since they were all related (we can't test python 3.10 without design changes for broken packages, and we can't test the design changes without python 3.10 fixes because of dep conflicts)

Sure, it's not the best approach but I guess it's better than the alternatives.

I didn't suggest separate PRs though. I mostly wanted to say that I don't see any problems with the changes but at the same time I just lack knowledge about these packages to say all of them are correct.

Perhaps you could squash the Openroad and `Build.yml` changes into the original commit. Otherwise LGTM. @umarcor ?

**Comment by proppy** [comment]
rebased, and squashed!

Thanks @ajelinski for the quick review.

**Pull request merged**